src/apirequest.ts

```diff
@@ -211,18 +187,8 @@ async function createAPIRequestAsync<T>(parameters: APIRequestParams) {
     return qs.stringify(params, {arrayFormat: 'repeat'});
   };
 
-  // delete path parameters from the params object so they do not end up in
-  // query
-  parameters.pathParams.forEach(param => {
-    delete params[param];
-    if (
-      parameters.context &&
-      parameters.context._options &&
-      parameters.context._options.params
-    ) {
-      delete parameters.context._options.params[param];
```

Comment:
Am I correct that this line here is the source of the bug, could this have just been removed?

Reply:
That's just the thing - I have no idea.  This code was so crufty, that it was getting really hard to understand the original intent.  I suspect this delete was needed, because otherwise path params were being included both in the URI with substituted, as well as in the querystring.  

Either way, I'm we needed to clean this up so it was at least comprehensible.